### PR TITLE
Add 'View project root' flag

### PR DESCRIPTION
**Pull request by tpasternak**
# Checklist

- [ ] I have filed an issue about this change and discussed potential changes with the maintainers.
- [ ] I have received the approval from the maintainers to make this change.
- [ ] This is not a stylistic, refactoring, or cleanup change.

Please note that the maintainers will not be reviewing this change until all checkboxes are ticked. See 
the [Contributions](https://github.com/bazelbuild/intellij#contributions) section in the README for more 
details.

# Discussion thread for this change

Issue number: `<please reference the issue number or url here>`

# Description of this change

https://github.com/bazelbuild/intellij/assets/1066003/29720ae3-bc31-4b05-b9d6-da26d2e9b760

### CODE REVIEW

File: base/src/com/google/idea/blaze/base/projectview/section/sections/ViewProjectRootSection.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.idea.blaze.base.projectview.section.sections;
+
+import com.google.idea.blaze.base.projectview.section.ScalarSection;
+import com.google.idea.blaze.base.projectview.section.SectionKey;
+import com.google.idea.blaze.base.projectview.section.SectionParser;
+
+/** If set to true, automatically derives targets from the project directories. */
```
Comment (blorente) [inline base/src/com/google/idea/blaze/base/projectview/section/sections/ViewProjectRootSection.java:22]: I think this comment is out of date.

File: base/tests/unittests/com/google/idea/blaze/base/projectview/ProjectViewSetTest.java [hunk 2]
```diff
@@ -120,6 +121,7 @@ public void testProjectViewSetSerializable() {
                         ScalarSection.builder(BuildConfigSection.KEY)
                             .set(new WorkspacePath("test")))
                     .add(ScalarSection.builder(UseExclusionPatternsSection.KEY).set(false))
+                    .add(ScalarSection.builder(ViewProjectRootSection.KEY).set(false))
```
Comment (blorente) [inline base/tests/unittests/com/google/idea/blaze/base/projectview/ProjectViewSetTest.java:124]: Could you add a test where we actually set it in a project view? We don't have to assert anything on the actual behaviour, I just want to check that it parses.

**Review by blorente — APPROVED** [review]
Approved with very minor comments! Thanks a lot for this change.

**Comment by tpasternak** [comment]
Oh, I can't just test it properly with the current test framework, because it converts `TestFileSystem`'s files to NIO files via paths without hestitation, so as a result the tests would crash because it's impossible to call `listFiles` on it. I removed the comment (as it's already commented in string) and I'll try to think about fixing the test framework later

https://github.com/bazelbuild/intellij/blob/master/base/tests/utils/integration/com/google/idea/blaze/base/BlazeIntegrationTestCase.java#L110

**Comment by blorente** [comment]
That's fair, it's fine to leave the proper tests as a follow up. Do you think it would be hard to add a test that asserts that the section parses correctly? So a test that takes the string:
```
view_project_root: true
```
And asserts that the right Section is created.

**Comment by tpasternak** [comment]
Ok, that's easy, could you take a look again?

**Comment by blorente** [comment]
Yeah, looks good to me! Thanks for this change.

**Pull request merged**